src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Models/AnalysisServicesClient.cs

```diff
@@ -87,6 +88,16 @@ public AnalysisServicesClient()
                 }
             }
 
+            GatewayDetails gatewayDetails = null;
+            if (gatewayResourceId == "-")
```

Comment:
I would like to see this special value, namely "-" defined in a common file as a constant in some common codethat can be comsumed from the various places this is used.